lib/audit_help.c

```diff
@@ -68,6 +68,39 @@ void audit_logger (int type, MAYBE_UNUSED const char *pgname, const char *op,
 	}
 }
 
+/*
+ * This function will log a message to the audit system using a predefined
+ * message format. Parameter usage is as follows:
+ *
+ * type - type of message: AUDIT_USER_MGMT for changing any account
+ *	  attributes.
+ * pgname - program's name
+ * op  -  operation. "adding user", "changing finger info", "deleting group"
+ * name - user's account or group name. If not available use NULL.
+ * id  -  uid or gid that the operation is being performed on. This is used
+ *	  only when user is NULL.
+ * grp - group name associated with event
+ */
+void audit_logger_with_group (int type, MAYBE_UNUSED const char *pgname,
+		const char *op, const char *name, unsigned int id,
+		const char *grp, shadow_audit_result result)
+{
+	int len;
+	char enc_group[(GROUP_NAME_MAX_LENGTH*2)+1], buf[1024];
+	if (audit_fd < 0) {
+		return;
+	}
+	len = strnlen(grp, sizeof(enc_group)/2);
+	if (audit_value_needs_encoding(grp, len)) {
+		snprintf(buf, sizeof(buf), "%s grp=%s", op,
+			audit_encode_value(enc_group, grp, len));
+	} else {
+		snprintf(buf, sizeof(buf), "%s grp=\"%s\"", op, grp);
```

Comment:
These should call SNPRINTF().

```c
$ grepc SNPRINTF .
./lib/string/sprintf/snprintf.h:#define SNPRINTF(s, fmt, ...)                                                 \
(                                                                             \
	snprintf_(s, NITEMS(s), fmt __VA_OPT__(,) __VA_ARGS__)                \
)
```

Reply:
Updated. Thanks! I suspected I must be missing something like this.